src/app/debug/table/table.component.ts

```diff
@@ -371,9 +373,13 @@ export class TableComponent implements OnInit, OnDestroy {
   }
 
   openReportInTab(): void {
-    let reportTab = this.tableSettings.reportMetadata.find((report) => report.checked);
+    let reportTab: Report | undefined = this.tableSettings.reportMetadata.find((report) => report.checked);
+    if (!reportTab) {
+      this.toastService.showDanger('Could not find report', 'No report found that was selected.');
+      return;
+    }
     this.httpService
-      .getReport(reportTab.storageId, this.viewSettings.currentView.storageName)
+      .getReport(String(reportTab.storageId), this.viewSettings.currentView.storageName)
```

Comment:
I see you casting storage id to a string a lot of times, I also see you've change the type in the interface from a number to a string, either change the method input parameters to accept a string, or keep it a number.
I think that if it can be a number then it should be a number.

Reply:
I tried getting rid of the string casting at all places where I could, but removing the string casting in tab.service.ts seems to break the debug_fileupload E2E test and I'm still not sure why at this point so I opted to leave it in that service for now.